src/core/main.c

```diff
@@ -3021,16 +3015,23 @@ int main(int argc, char *argv[]) {
                 goto finish;
         }
 
-        (void) invoke_main_loop(m,
-                                &saved_rlimit_nofile,
-                                &saved_rlimit_memlock,
-                                &reexecute,
-                                &retval,
-                                &shutdown_verb,
-                                &fds,
-                                &switch_root_dir,
-                                &switch_root_init,
-                                &error_message);
+        r = invoke_main_loop(m,
```

Comment:
so up-top you changed invoke_main_loop() to return ManagerObjective, and now you store the result in an `int` anyway?

I'd be more comfortable with returning *any* error numbers as ManagerObjective, if the enum had a line `MANAGER_OBJECTIVE_MIN = -ERRNO_MAX` or so, to underline that not just EINVAL but in fact *any* errno can be stored in it.

Reply:
And I'd probably introduce a local variable "objective" here too and not use `r` for this anymore. Otherwise it just seems too likely that we'll accidentally override `r` with some other failure but this failure is too important for that, I think it deserves its own properly named local variable.

Reply:
Here `r` is used throughout the function to store the state (with `goto finish`). So if this variable was renamed, we wouldn't have `r` but only this new variable. I think this would be almost the same, but we'd have a longer and unusual variable name.